server/src/main/java/io/crate/execution/ddl/tables/CreateIndexTask.java

```diff
@@ -0,0 +1,292 @@
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.*;
+
+import static io.crate.execution.ddl.tables.AddColumnTask.stateAndMapping;
+import static org.elasticsearch.cluster.metadata.IndexMetadata.*;
+import static org.elasticsearch.cluster.metadata.MetadataCreateIndexService.*;
+import static org.elasticsearch.cluster.metadata.MetadataCreateIndexService.IndexCreationTask.ensureTableInfoCanBeCreated;
+
+/**
+ * Hybrid of:
+ * - {@link MetadataCreateIndexService.IndexCreationTask} to validate settings, create/remove probe index and add index metadata into cluster state.
+ * - {@link AddColumnTask} to get mapping change
+ *
+ * IndexCreationTask is taken without handling of templates, aliases, resize, request.blocks (which is always empty anyway) and CLOSED state.
+ * as it's all irrelevant for creation of non-partitioned tables.
+ */
+public class CreateIndexTask extends DDLClusterStateTaskExecutor<CreateTableRequest> {
+
+    private static final Logger LOGGER = LogManager.getLogger(CreateIndexTask.class);
+
+    private final NodeContext nodeContext;
+    private final IndicesService indicesService;
+    private final boolean forbidPrivateIndexSettings;
+    private final Settings settings;
+    private final IndexScopedSettings indexScopedSettings;
+    private final ShardLimitValidator shardLimitValidator;
+    private final AllocationService allocationService;
+    private final Environment env;
+
+    public CreateIndexTask(NodeContext nodeContext,
+                           IndicesService indicesService,
+                           Settings settings,
+                           IndexScopedSettings indexScopedSettings,
+                           AllocationService allocationService,
+                           Environment env,
+                           ShardLimitValidator shardLimitValidator,
+                           boolean forbidPrivateIndexSettings) {
+        this.nodeContext = nodeContext;
+        this.indicesService = indicesService;
+        this.settings = settings;
+        this.indexScopedSettings = indexScopedSettings;
+        this.allocationService = allocationService;
+        this.env = env;
+        this.shardLimitValidator = shardLimitValidator;
+        this.forbidPrivateIndexSettings = forbidPrivateIndexSettings;
+    }
+
+    @Override
+    protected ClusterState execute(ClusterState currentState, CreateTableRequest request) throws Exception {
+        Index createdIndex = null;
+        String removalExtraInfo = null;
+        IndicesClusterStateService.AllocatedIndices.IndexRemovalReason removalReason = IndicesClusterStateService.AllocatedIndices.IndexRemovalReason.FAILURE;
+        try {
+            validate(request.getTableName().indexNameOrAlias(), settings, currentState);
+
+            Map<String, Object> mapping = stateAndMapping(currentState,
+                                                          request.addColumnRequest(),
```

Comment:
CREATE TABLE used to operate with mapping, now uses CrateDb specific structures (Reference/AddColumnRequest) so this is replacement for 

```
Map<String, Object> mapping;
        String mappingStr = request.mapping();
        if (mappingStr == null) {
            mapping = new HashMap<>();
        } else {
            mapping = MapperService.parseMapping(xContentRegistry, mappingStr);
        }
```